chain/near/src/chain.rs

```diff
@@ -0,0 +1,383 @@
+                        ),
+                        number: header.height as i32,
+                    },
+                    Some(response.cursor.clone()),
+                ))
+            }
+
+            bstream::ForkStep::StepIrreversible => {
+                panic!("irreversible step is not handled and should not be requested in the Firehose request")
+            }
+
+            bstream::ForkStep::StepUnknown => {
+                panic!("unknown step should not happen in the Firehose response")
+            }
+        }
+    }
+}
+
+impl FirehoseMapper {
+    // FIXME: This should be replaced by using the `TriggersAdapter` struct directly. However, the TriggersAdapter trait
+    //        is async. It's actual async usage is done inside a manual `poll` implementation in `firehose_block_stream#poll_next`
+    //        value. An upcoming improvement will be to remove this `poll_next`. Once the refactor occurs, this should be
+    //        removed and TriggersAdapter::triggers_in_block should be use straight.
+    fn firehose_triggers_in_block(
+        &self,
+        block: &codec::BlockWrapper,
+        _filter: &TriggerFilter,
+    ) -> Result<BlockWithTriggers<Chain>, FirehoseError> {
+        let block = block.block.as_ref().unwrap();
+        let header = block.header.as_ref().unwrap();
+        let near_block = NearBlock {
+            hash: H256::from_slice(&header.hash.as_ref().unwrap().bytes.clone()),
+            number: header.height,
+            parent_hash: header
+                .prev_hash
+                .as_ref()
+                .map(|v| H256::from_slice(&v.bytes.clone())),
+            parent_number: header.prev_hash.as_ref().map(|_| header.prev_height),
+        };
+        let block_ptr = BlockPtr::from(&near_block);
+
+        Ok(BlockWithTriggers {
+            block: near_block,
+            trigger_data: vec![NearTrigger::Block(block_ptr, NearBlockTriggerType::Every)],
+        })
+    }
+}
+
+pub struct IngestorAdapter {
+    logger: Logger,
+}
+
+#[async_trait]
+impl IngestorAdapterTrait<Chain> for IngestorAdapter {
+    fn logger(&self) -> &Logger {
+        &self.logger
+    }
+
+    fn ancestor_count(&self) -> BlockNumber {
+        0
+    }
+
+    async fn latest_block(&self) -> Result<BlockPtr, IngestorError> {
+        Ok(BlockPtr {
+            hash: BlockHash::from(vec![0xff; 32]),
+            number: 0,
+        })
+    }
+
+    async fn ingest_block(
+        &self,
+        _block_hash: &BlockHash,
+    ) -> Result<Option<BlockHash>, IngestorError> {
+        // FIXME (NEAR):  Might not be necessary for NEAR support for now
+        Ok(None)
+    }
+
+    fn chain_head_ptr(&self) -> Result<Option<BlockPtr>, Error> {
+        // FIXME (NEAR):  Might not be necessary for NEAR support for now
+        Ok(None)
+    }
+
+    fn cleanup_cached_blocks(&self) -> Result<Option<(i32, usize)>, Error> {
+        // FIXME (NEAR):  Might not be necessary for NEAR support for now
```

Comment:
An ingestor will hopefully be unnecessary in a firehose setup. We can later find a way to reduce the boilerplate.